### PR TITLE
Enforce Max Committee Size and Remove Multiple Slots Per Committee 

### LINKED ISSUE #579: Enforce Max Committees per Slot
`getCommitteeParams` has a bug. The max number of committees per slot should be bounded by the total number of shards. The max committees per slot should also be bounded by total number of shards divide cycle length.

From 2.1 spec:
```
min(len(avs) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1, SHARD_COUNT // CYCLE_LENGTH)
```

Here is the line that needs to change:
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/casper/sharding.go#L62

### LINKED ISSUE #639: Remove Multiple Slots Per Committee for New Shuffling
As discussed in the last implementers call, the rationale of having a single committee covers multiple slots only appears when validator count is less than CYCLE_LENGTH * MIN_COMMITEE which is 8192 given current spec. In this case, we'll be using a smaller committee instead.

https://github.com/ethereum/eth2.0-specs/pull/62

### PR DESCRIPTION
**Pull request by IvanTheGreatDev**
This is to enforce the max committee size as per the spec as of 2.1 and to remove multiple slots per committee as they aren't a benefit to security.

resolves #579 
resolves #639

### CODE REVIEW

File: beacon-chain/casper/sharding.go [hunk 1]
```diff
@@ -53,21 +52,14 @@ func splitBySlotShard(shuffledValidators []uint32, crosslinkStartShard uint64) [
...
-// to attest the same shard.
-func getCommitteeParams(numValidators int) (committeesPerSlot, slotsPerCommittee int) {
-	if numValidators >= int(params.GetConfig().CycleLength*params.GetConfig().MinCommiteeSize) {
-		committeesPerSlot := numValidators/int(params.GetConfig().CycleLength*params.GetConfig().MinCommiteeSize*2) + 1
-		return committeesPerSlot, 1
+// If numActiveValidators > CycleLength * MinCommitteeSize, committees are based off a max amount
+// of len(avs) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1 or SHARD_COUNT // CYCLE_LENGTH
+func getCommitteeParams(numValidators int) (committeesPerSlot int) {
+	cycleLength := int(params.GetConfig().CycleLength)
+	boundOnAVS := numValidators/cycleLength/int(params.GetConfig().MinCommiteeSize*2) + 1
```
Comment (rauljordan) [inline beacon-chain/casper/sharding.go]: what is AVS? Can you explain with some comments?

File: beacon-chain/casper/sharding.go [hunk 2]
```diff
@@ -53,21 +52,14 @@ func splitBySlotShard(shuffledValidators []uint32, crosslinkStartShard uint64) [
...
-// If numActiveValidators > CycleLength * MinCommitteeSize, multiple committees are selected
-// to attest the same shard in a single slot.
-// If numActiveValidators < CycleLength * MinCommitteeSize, committees span across multiple slots
-// to attest the same shard.
-func getCommitteeParams(numValidators int) (committeesPerSlot, slotsPerCommittee int) {
-	if numValidators >= int(params.GetConfig().CycleLength*params.GetConfig().MinCommiteeSize) {
-		committeesPerSlot := numValidators/int(params.GetConfig().CycleLength*params.GetConfig().MinCommiteeSize*2) + 1
-		return committeesPerSlot, 1
+// If numActiveValidators > CycleLength * MinCommitteeSize, committees are based off a max amount
+// of len(avs) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1 or SHARD_COUNT // CYCLE_LENGTH
```
Comment (terenc3t) [inline beacon-chain/casper/sharding.go]: Nitpick, can we write out the sentence instead python pseudocode? pseudocode makes it hard to understand

File: beacon-chain/casper/sharding.go (hunk 1 above)
Comment (IvanTheGreatDev) [reply beacon-chain/casper/sharding.go]: Sorry! AVS is active validators, I'll take note of that.

File: beacon-chain/casper/sharding.go (hunk 2 above)
Comment (IvanTheGreatDev) [reply beacon-chain/casper/sharding.go]: Sure thing!

**Pull request merged**